packages/mainmenu-extension/src/index.ts

```diff
@@ -535,19 +535,23 @@ function createKernelMenu(
   commands.addCommand(CommandIDs.shutdownAllKernels, {
     label: trans.__('Shut Down All Kernels…'),
     isEnabled: () => {
-      return !app.serviceManager.sessions.running().next().done;
+      return !app.serviceManager.kernels.running().next().done;
     },
     execute: () => {
       return showDialog({
         title: trans.__('Shut Down All?'),
-        body: trans.__('Shut down all kernels?'),
+        body: trans._n(
+          'Are you sure you want to permanently shut down the running kernel?',
+          'Are you sure you want to permanently shut down the %1 running kernels?',
+          app.serviceManager.kernels.runningCount
+        ),
         buttons: [
-          Dialog.cancelButton({ label: trans.__('Dismiss') }),
+          Dialog.cancelButton(),
           Dialog.warnButton({ label: trans.__('Shut Down All') })
         ]
       }).then(result => {
         if (result.button.accept) {
-          return app.serviceManager.sessions.shutdownAll();
+          return app.serviceManager.kernels.shutdownAll();
```

Comment:
I think we want to retain the old behaviour of calling on `sessions.shutdownAll()` here. I am not quite sure of implications if we were to change it so I would not touch it. @afshin may have an opinion as he's worked on the adjacent code recently.

```suggestion
          return app.serviceManager.sessions.shutdownAll();
```

Reply:
I think @krassowski is right and we should keep `sessions.shutdownAll()`. Sessions are owned by the JupyterLab server, but kernels may have external origins.

Relatedly, this conversation highlights an issue that the label for this command is `Shut Down All Kernels…` and this ambiguity is unfortunate. We don't have an end-user concept of "session" well-defined, but we should think about the language of this command.

Reply:
Thanks for the feedback, @afshin!

Given this feedback, I have two questions: 

- In the sidebar, the `shutdownAll` action is triggered for the `kernels` (see snippet below, please). Should we change it to `sessions` or both?
- In the case of the body text for the command palette dialog, should the kernel count be a session count instead? Should I implement `runningCount` in the session manager?

Thanks!

https://github.com/jupyterlab/jupyterlab/blob/653c22621f996249862093d0bb0f23b1279b0f52/packages/running-extension/src/kernels.tsx#L88

Reply:
I'm not really sure what to do here. The kernels panel very clearly shows actual kernels, including kernels that do not have an actual "session" associated with them. So that's clear at least.

But as far as the shut down links for them and the running count, I think we should think about the user experience a bit and make some decisions on whether we need to solve this ambiguity or leave things as they are. And if we do need to solve them, the answer might be different language or it might be different behavior. I'm a little unsure, personally.

Reply:
Thanks for sharing your perspective, @afshin!

So, maybe this PR should just be to standardize the body text of the two dialogs without implementing `runningCount`?